src/Storage/Snowflake/SnowflakeImportAdapter.php

```diff
@@ -56,14 +58,18 @@ public function runCopyCommand(
             implode(', ', $quotedColumns)
         );
 
-        $sql .= sprintf(
-            ' SELECT %s FROM %s.%s',
-            implode(', ', $quotedColumns),
-            $this->connection->quoteIdentifier($source->getSchema()),
-            $this->connection->quoteIdentifier($source->getTableName())
-        );
-
-        $this->connection->query($sql);
+        if ($source instanceof Table) {
+            $sql .= sprintf(
+                ' SELECT %s FROM %s.%s',
+                implode(', ', $quotedColumns),
+                $this->connection->quoteIdentifier($source->getSchema()),
+                $this->connection->quoteIdentifier($source->getTableName())
+            );
+            $this->connection->query($sql);
+        } elseif ($source instanceof SelectSource) {
```

Comment:
Tohle mi přijde takový divný, čekal bych že pro oba typy zdrojů `Table` i `SelectSource` bude ten kód stejný jako pro `SelectSource`.
Ten `getFromStatement` pro tabulku vrací např `"in.c-main"."some-table"` a pro `SelectSource` např. `select * from "in.c-main"."some-table"`. Nemělo by to vracet stejný typ dotazu, nebo neměly by to být různé metody?

Reply:
Nad tím sem přemýšlel je tam ten `getFromStatement` chtěl sem tam navíc vyblít ty sloupce, protože ten `SelectSource` když sa sestaví v connection tak ten seznam sloupců bude vždycky mět. Přemýšlal sem, že bych columns přehodil z `ImportOptions` do `Table` tam by dávaly možná větší smysl.

Reply:
Asi je to spíš drobnost. Vidím že tam kde se to používá pro exporty jako zdroj pro COPY command to podporuje oboje jak query tak identifikátor tabulky takže to co vrací `getFromStatement` je ok.

Pokud by se columns přesouvaly musel by je mít imho už `SourceInterface`.

Reply:
Stačí to v table, udělal sem na to vlastní PR, https://github.com/keboola/php-db-import-export/pull/29